src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,78 @@
 package core.basesyntax;
 
+import java.util.LinkedList;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private LinkedList<Entry<K, V>>[] table;
+    private int size;
 
+    public MyHashMap() {
+        table = new LinkedList[INITIAL_CAPACITY];
+        size = 0;
+    }
     @Override
     public void put(K key, V value) {
+        int index = getIndex(key);
+        if (table[index] == null) {
+            table[index] = new LinkedList<>();
+        }
+        for (Entry<K, V> entry : table[index]) {
+            if (entry.key.equals(key)) {
+                entry.value = value;
+                return;
+            }
+        }
+        table[index].add(new Entry<>(key, value));
+        size++;
 
+        if (size > LOAD_FACTOR * table.length) {
+            resize();
+        }
     }
-
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        if (table[index] != null) {
+            for (Entry<K, V> entry : table[index]) {
+                if (entry.key.equals(key)) {
+                    return entry.value;
+                }
+            }
+        }
         return null;
     }
-
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+    private int getIndex(K key) {
+        return Math.abs(key.hashCode()) % table.length;
```

Comment:
The `getIndex` method does not handle `null` keys. If `key` is `null`, calling `key.hashCode()` will throw a `NullPointerException`. Consider adding a check for `null` keys and decide how you want to handle them (e.g., throw an exception or use a special index).